drivers/clock_control/nrf5_power_clock.c

```diff
@@ -11,7 +11,9 @@
 #include <device.h>
 #include <clock_control.h>
 #include <misc/__assert.h>
+#if !defined(CONFIG_ARCH_POSIX)
```

Comment:
One trick you could do, is make sure the header guard that is in cmsis.h is defined when CONFIG_ARCH_POSIX. Then you don't have to express this for every user of cmsis.h.

Alternatively, modify the include paths and provide a stub cmsis.h.

Reply:
this include is not needed, it should be part of soc.h already. Simply remove it.